api/src/main/java/ru/urururu/util/FinalMap.java

```diff
@@ -0,0 +1,55 @@
+package ru.urururu.util;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * @author <a href="mailto:dmitriy.g.matveev@gmail.com">Dmitry Matveev</a>
+ */
+public class FinalMap {
+    public static <K, V> Map<K, V> createHashMap() {
+        return new HashMap<K, V>() {
+            @Override
+            public V get(Object key) {
+                return nonNull(key, super.get(key));
+            }
+
+            @Override
+            public V put(K key, V value) {
+                return validate(key, value, super.put(key, value));
+            }
+        };
+    }
+
+    public static <K, V> Map<K, V> createLinkedHashMap() {
+        return new LinkedHashMap<K, V>() {
+            @Override
+            public V get(Object key) {
+                return nonNull(key, super.get(key));
+            }
+
+            @Override
+            public V put(K key, V value) {
+                return validate(key, value, super.put(key, value));
+            }
+        };
+    }
+
+    private static <K, V> V nonNull(K key, V value) {
+        if (value == null) {
+            throw new IllegalStateException("No mapping for " + key);
+        }
+
+        return value;
+    }
+
+    private static <K, V> V validate(K key, V value, V oldValue) {
+        if (oldValue != null) {
+            // todo is it possible to support null values?
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)